File: GXpert/GXpert.Web/Modules/ServerTypes/Syllabus/SubjectColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SubjectRow } from "./SubjectRow";

export interface SubjectColumns {
    Id: Column<SubjectRow>;
    CourseName: Column<SubjectRow>;
    ClassTitle: Column<SubjectRow>;
    SemesterName: Column<SubjectRow>;
    Title: Column<SubjectRow>;
    SortOrder: Column<SubjectRow>;
    Weightage: Column<SubjectRow>;
    Thumbnail: Column<SubjectRow>;
    Description: Column<SubjectRow>;
}

export class SubjectColumns extends ColumnsBase<SubjectRow> {
    static readonly columnsKey = 'Syllabus.Subject';
    static readonly Fields = fieldsProxy<SubjectColumns>();
}

File: GXpert/GXpert.Web/Modules/Syllabus/Subject/SubjectColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Syllabus.Columns;

[ColumnsScript("Syllabus.Subject")]
[BasedOnRow(typeof(SubjectRow), CheckNames = true)]
public class SubjectColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [Width(100),QuickFilter]
    public string CourseName { get; set; }
    public string ClassTitle { get; set; }
    public string SemesterName { get; set; }
    public string Title { get; set; }
    public short SortOrder { get; set; }

    /* public string MediumTitle { get; set; }*/
    public float Weightage { get; set; }
    public string Thumbnail { get; set; }
    public string Description { get; set; }
   
}